src/modules/flight_mode_manager/tasks/Auto/FlightTaskAuto.cpp

```diff
@@ -810,6 +810,12 @@ void FlightTaskAuto::_updateTrajConstraints()
 	_position_smoothing.setMaxVelocityXY(_param_mpc_xy_vel_max.get());
 	_position_smoothing.setMaxJerk(_param_mpc_jerk_auto.get()); // TODO : Should be computed using heading
 
+	// Stretch the constraints of the velocity controller to leave some room for an additional
+	// correction required by the altitude/vertical position controller
+	_constraints.speed_down = 1.2f * _param_mpc_z_v_auto_dn.get();
+	_constraints.speed_up = 1.2f * _param_mpc_xy_vel_max.get();
+	_constraints.speed_xy = 1.2f * _param_mpc_xy_vel_max.get();;
```

Comment:
I moved this up here and removed the max, as there wasn't a possibility that the constraints to back to the normal value after emergency braking, I haven't experienced any issues stemming from this. But this way it should be more intentional.